tests/main_test.go

```
@@ -4,24 +4,90 @@ import (
 	"context"
 	"flag"
 	"os"
+	"os/signal"
+	"syscall"
 	"testing"
+	"unsafe"
+
+	"github.com/satori/go.uuid"
 )
 
+var testUUID string
+
+// TODO(dperny) hidden field, maintenance nightmare
+// Copied from testing.M
+type _M struct {
+	matchString func(pat, str string) (bool, error)
+	tests       []testing.InternalTest
+	benchmarks  []testing.InternalBenchmark
+	examples    []testing.InternalExample
+}
+
 func TestMain(m *testing.M) {
 	// gotta call this at the start or NONE of the flags work
 	flag.Parse()
 
+	// get the tests
+	tests := (*_M)(unsafe.Pointer(m)).tests
```

I would advise against this sort of wizardry.

When it fails, it fails hard